### ISSUE TITLE
Hooks not working when using a custom resolver that extends CRUDResolver.

### ISSUE THREAD

**Issue by luca-nardelli** [doug-martin/nestjs-query#957]
**Describe the bug**
When using a custom resolver (extending CRUDResolver) and hooks (@BeforeCreateOne, etc...), the hooks do not work.

### Have you read the [Contributing Guidelines](https://github.com/doug-martin/nestjs-query/blob/master/CONTRIBUTING.md)?
Yes

**To Reproduce**
* Clone my fork https://github.com/luca-nardelli/nestjs-query `hooks` branch
* Start docker containers
* Run tests
  * `npm run jest:e2e examples/hooks/e2e/todo-item.resolver.spec.ts` runs the original tests with the auto-generated resolver
  * `npm run jest:e2e examples/hooks/e2e/todo-item-cr.resolver.spec.ts` runs the same test suite with a custom resolver that extends CRUDResolver.

**Expected behavior**
Tests should pass for both situations.

**Screenshots**
Not a screenshot, but one of the 4 failing tests is the following
```
  ● TodoItemResolver (hooks - e2e) › update many › should call the beforeUpdateMany hook when updating todoItem

    expect(received).toEqual(expected) // deep equality

    - Expected  - 2
    + Received  + 2

      Array [
        Object {
          "completed": true,
          "id": 10,
          "title": "Update Many Hook",
    -     "updatedBy": "e2e@nestjs-query.com",
    +     "updatedBy": null,
        },
        Object {
          "completed": true,
          "id": 11,
          "title": "Update Many Hook",
    -     "updatedBy": "e2e@nestjs-query.com",
    +     "updatedBy": null,
        },
      ]
```

**Desktop (please complete the following information):**
 - Node Verson [e.g. 14.9.0]
 - Nestjs-query Version [latest]

**Additional context**
The issue seems to disappear if I edit line 35 of `hook.interceptor.ts` as follows: `return this.moduleRef.get<Hook<unknown>>(hookToken, {strict: false});`
This probably means it's a nestjs provider scoping issue, but I'm not sure this could be the fix, since it could lead to token clashes.

I could make a PR to add the new test scenario, but I wouldn't know how to fix the issue at hand at the moment (apart from the strict: false thing). Moreover, it seems to me that there is useless duplication of the test files for the auto-resolver and custom-resolver scenarios. Does anyone know of a more elegant way to run the same batch of tests with a different appmodule configuration?

**Comment by doug-martin** [comment]
@luca-nardelli sorry this took me a while to look at. I think I see what the problem is, the hook is registered on the `TodoItemUpdateDTO` can you try changing the following 

```ts
 @UseInterceptors(HookInterceptor(HookTypes.BEFORE_UPDATE_MANY, TodoItemDTO))
```

To 

```ts
 @UseInterceptors(HookInterceptor(HookTypes.BEFORE_UPDATE_MANY, TodoItemUpdateDTO))
```

Let me know if that works (or doesn't) and I can help debug further.

**Comment by luca-nardelli** [comment]
Hey, no problem, for now I worked around the issue by doing what I wrote in the "additional context" (but it's an ugly workaround!)

I've tried editing the UseInterceptors but unfortunately it didn't change anything. I would say it makes sense though, as the failing tests involve the auto-generated mutations (update one/update many) and the only usage of UseInterceptors I found was related to the additional mutation `markTodoItemsAsCompleted`

You can see what I did here: https://github.com/luca-nardelli/nestjs-query/blob/hooks/examples/hooks/src/todo-item/todo-item.resolver.ts

Basically I've defined a custom resolver that simply extends the CRUD resolver (without defining any additional methods), and I've set up the module so that it can either use the auto-generated resolver or this one according to its configuration (in order to test both with Jest), see here for details https://github.com/luca-nardelli/nestjs-query/blob/hooks/examples/hooks/src/todo-item/todo-item.module.ts

I think, but I'm not sure about it, that since the resolver is defined in the scope of todo-item module, Nest cannot find the hook interceptor (when looking for it https://github.com/luca-nardelli/nestjs-query/blob/hooks/packages/query-graphql/src/interceptors/hook.interceptor.ts#L35) since that is created in the scope of NestjsQueryGraphQLModule.forFeature
By having a look here (https://github.com/nestjs/nest/issues/832) it seems that without `strict: false` `moduleRef.get()` only looks in the current module, even ignoring imported providers

**Comment by luca-nardelli** [comment]
By the way, this is the full log for the failing tests 

```
> jest --runInBand --coverage --config=./jest.e2e.config.js "examples/hooks/e2e/todo-item-cr.resolver.spec.ts"

 FAIL   E2E  examples/hooks/e2e/todo-item-cr.resolver.spec.ts (7.96 s)
  TodoItemResolver (hooks - e2e)
    find one
      ✓ should find a todo item by id (43 ms)
      ✓ should return null if the todo item is not found (8 ms)
      ✓ should return subTasks as a connection (19 ms)
      ✓ should return tags as a connection (11 ms)
    query
      ✓ should return a connection (8 ms)
      ✓ should allow querying (9 ms)
      ✓ should allow querying on subTasks (9 ms)
      ✓ should allow querying on tags (11 ms)
      ✓ should allow sorting (8 ms)
      paging
        ✓ should allow paging with the 'first' field (6 ms)
        ✓ should allow paging with the 'first' field and 'after' (9 ms)
    create one
      ✓ should require a header secret (6 ms)
      ✓ should allow creating a todoItem (10 ms)
      ✕ should call the beforeCreateOne hook (12 ms)
      ✓ should validate a todoItem (7 ms)
    create many
      ✓ should require a header secret (3 ms)
      ✓ should allow creating multiple todoItems (8 ms)
      ✕ should call the beforeCreateMany hook when creating multiple todoItems (8 ms)
      ✓ should validate a todoItem (4 ms)
    update one
      ✓ should require a header secret (3 ms)
      ✓ should allow updating a todoItem (10 ms)
      ✕ should call the beforeUpdateOne hook (9 ms)
      ✓ should require an id (14 ms)
      ✓ should validate an update (8 ms)
    update many
      ✓ should require a header secret (4 ms)
      ✓ should allow updating a todoItem (9 ms)
      ✕ should call the beforeUpdateMany hook when updating todoItem (13 ms)
      ✓ should require a filter (6 ms)
      ✓ should require a non-empty filter (8 ms)
    delete one
      ✓ should require a header secret (3 ms)
      ✓ should allow deleting a todoItem (11 ms)
      ✓ should require an id (3 ms)
    delete many
      ✓ should require a header secret (4 ms)
      ✓ should allow updating a todoItem (6 ms)
      ✓ should require a filter (3 ms)
      ✓ should require a non-empty filter (4 ms)
    addSubTasksToTodoItem
      ✓ allow adding subTasks to a todoItem (15 ms)
    addTagsToTodoItem
      ✓ allow adding subTasks to a todoItem (12 ms)
    removeTagsFromTodoItem
      ✓ allow adding subTasks to a todoItem (14 ms)

  ● TodoItemResolver (hooks - e2e) › create one › should call the beforeCreateOne hook

    expected {
      data: {
        createOneTodoItem: {
          id: '7',
          title: 'Create One Hook Todo',
          completed: false,
          createdBy: 'e2e@nestjs-query.com'
        }
      }
    } response body, got {
      data: {
        createOneTodoItem: {
          id: '7',
          title: 'Create One Hook Todo',
          completed: false,
          createdBy: null
        }
      }
    }

      448 |         }`,
      449 |         })
    > 450 |         .expect(200, {
          |          ^
      451 |           data: {
      452 |             createOneTodoItem: {
      453 |               id: '7',

      at Object.<anonymous> (examples/hooks/e2e/todo-item-cr.resolver.spec.ts:450:10)
      ----
      at error (node_modules/supertest/lib/test.js:329:13)
      at Test.Object.<anonymous>.Test._assertBody (node_modules/supertest/lib/test.js:233:14)
      at node_modules/supertest/lib/test.js:80:15
      at Test.Object.<anonymous>.Test._assertFunction (node_modules/supertest/lib/test.js:311:11)
      at Test.Object.<anonymous>.Test.assert (node_modules/supertest/lib/test.js:201:21)
      at Server.localAssert (node_modules/supertest/lib/test.js:159:12)

  ● TodoItemResolver (hooks - e2e) › create many › should call the beforeCreateMany hook when creating multiple todoItems

    expected { data: { createManyTodoItems: [ [Object], [Object] ] } } response body, got { data: { createManyTodoItems: [ [Object], [Object] ] } }

      567 |         }`,
      568 |         })
    > 569 |         .expect(200, {
          |          ^
      570 |           data: {
      571 |             createManyTodoItems: [
      572 |               { id: '10', title: 'Many Create Hook 1', completed: false, createdBy: 'e2e@nestjs-query.com' },

      at Object.<anonymous> (examples/hooks/e2e/todo-item-cr.resolver.spec.ts:569:10)
      ----
      at error (node_modules/supertest/lib/test.js:329:13)
      at Test.Object.<anonymous>.Test._assertBody (node_modules/supertest/lib/test.js:233:14)
      at node_modules/supertest/lib/test.js:80:15
      at Test.Object.<anonymous>.Test._assertFunction (node_modules/supertest/lib/test.js:311:11)
      at Test.Object.<anonymous>.Test.assert (node_modules/supertest/lib/test.js:201:21)
      at Server.localAssert (node_modules/supertest/lib/test.js:159:12)

  ● TodoItemResolver (hooks - e2e) › update one › should call the beforeUpdateOne hook

    expected {
      data: {
        updateOneTodoItem: {
          id: '7',
          title: 'Update One Hook Todo',
          completed: true,
          updatedBy: 'e2e@nestjs-query.com'
        }
      }
    } response body, got {
      data: {
        updateOneTodoItem: {
          id: '7',
          title: 'Update One Hook Todo',
          completed: true,
          updatedBy: null
        }
      }
    }

      680 |         }`,
      681 |         })
    > 682 |         .expect(200, {
          |          ^
      683 |           data: {
      684 |             updateOneTodoItem: {
      685 |               id: '7',

      at Object.<anonymous> (examples/hooks/e2e/todo-item-cr.resolver.spec.ts:682:10)
      ----
      at error (node_modules/supertest/lib/test.js:329:13)
      at Test.Object.<anonymous>.Test._assertBody (node_modules/supertest/lib/test.js:233:14)
      at node_modules/supertest/lib/test.js:80:15
      at Test.Object.<anonymous>.Test._assertFunction (node_modules/supertest/lib/test.js:311:11)
      at Test.Object.<anonymous>.Test.assert (node_modules/supertest/lib/test.js:201:21)
      at Server.localAssert (node_modules/supertest/lib/test.js:159:12)

  ● TodoItemResolver (hooks - e2e) › update many › should call the beforeUpdateMany hook when updating todoItem

    expect(received).toEqual(expected) // deep equality

    - Expected  - 2
    + Received  + 2

      Array [
        Object {
          "completed": true,
          "id": 10,
          "title": "Update Many Hook",
    -     "updatedBy": "e2e@nestjs-query.com",
    +     "updatedBy": null,
        },
        Object {
          "completed": true,
          "id": 11,
          "title": "Update Many Hook",
    -     "updatedBy": "e2e@nestjs-query.com",
    +     "updatedBy": null,
        },
      ]

      831 |               updatedBy: ti.updatedBy,
      832 |             })),
    > 833 |           ).toEqual([
          |             ^
      834 |             { id: 10, title: 'Update Many Hook', completed: true, updatedBy: 'e2e@nestjs-query.com' },
      835 |             { id: 11, title: 'Update Many Hook', completed: true, updatedBy: 'e2e@nestjs-query.com' },
      836 |           ]);

      at examples/hooks/e2e/todo-item-cr.resolver.spec.ts:833:13

```

**Comment by doug-martin** [comment]
@luca-nardelli this is fixed in `@nestjs-query/query-graphql` v0.24.2. I ended up changing the interceptor to look up the correct hook token and using @Inject instead of using `ModuleRef`. Thanks again for submitting an issue!
